synapse/lib/cell.py

```diff
@@ -2628,7 +2653,19 @@ async def addHttpsPort(self, port, host='0.0.0.0', sslctx=None):
         }
         serv = self.wapp.listen(port, address=addr, ssl_options=sslctx, **kwargs)
         self.httpds.append(serv)
-        return list(serv._sockets.values())[0].getsockname()
+
+        lhost, lport = list(serv._sockets.values())[0].getsockname()
+
+        if managed_ctx:
+            # If the Cell is managing the SSLCtx, then we register a reload handler
+            # for it which reloads the pkey / cert.
+            def reload():
+                sslctx.load_cert_chain(certpath, pkeypath)
+                return True
+
+            self.addReloadableSystem(f'HTTPS_SSL_Reload_Cert_port={lport}', reload)
```

Comment:
Def do not want to require a-priori configuration knowledge to trigger this reload. `https:certs` should be a fine subsystem name. :)